easybuild/framework/easyconfig/easyconfig.py

```diff
@@ -151,14 +156,14 @@ def get_toolchain_hierarchy(parent_toolchain):
         # grab the easyconfig of the current toolchain and search the dependencies for a version of the subtoolchain
         path = robot_find_easyconfig(current_tc_name, current_tc_version)
         if path is None:
-            raise EasyBuildError("Could not find easyconfig for %(name)s toolchain version %(version)s",
+            raise EasyBuildError("Could not find easyconfig for %s toolchain version %s",
                                  current_tc_name, current_tc_version)
 
         # parse the easyconfig
         parsed_ec = process_easyconfig(path, validate=False)[0]
 
         # search the toolchain+dependencies for the version of the subtoolchain
-        dep_tcs = []
+        dep_tcs = [d for d in parsed_ec['dependencies'] if d['name'] == subtoolchain_name]
```

Comment:
I know this is a little pedantic since we have solved the particular problem with `icc/ifort` but...
Originally this function only dealt with toolchains from the depenencies, now it is dealing with dependencies *and* the toolchains of dependencies but these are not the same. Dependencies can have a `versionsuffix` which toolchains don't, so if we are adding dependencies we need to add them as
```
{'name': d['name'], ''version': '%s%s' % (d['version'], d['versionsuffix'])}
```
Same is true when adding dependencies below (basically need a repeat of this line). And I still think we should also be adding toolchains of deps of deps, since we've already parsed the deps so it comes at zero cost.